pkg/apis/kubeone/v1beta2/types.go

```diff
@@ -72,10 +72,38 @@ type ContainerRuntimeConfig struct {
 }
 
 // ContainerRuntimeDocker defines docker container runtime
-type ContainerRuntimeDocker struct{}
+type ContainerRuntimeDocker struct {
+	RegistryMirrors []string `json:"registryMirrors"`
+}
 
 // ContainerRuntimeContainerd defines docker container runtime
-type ContainerRuntimeContainerd struct{}
+type ContainerRuntimeContainerd struct {
+	Registry ContainerdRegistry `json:"registry,omitempty"`
+}
+
+type ContainerdRegistry struct {
+	Mirrors map[string]ContainerdRegistryMirror `json:"mirrors,omitempty"`
+	Configs map[string]ContainerdRegistryConfig `json:"configs,omitempty"`
+}
+
+type ContainerdRegistryMirror struct {
+	Endpoint []string `json:"endpoint,omitempty"`
+}
+
+type ContainerdRegistryConfig struct {
+	TLS  ContainerdRegistryConfigTLS  `json:"tls,omitempty"`
+	Auth ContainerdRegistryConfigAuth `json:"auth,omitempty"`
+}
+
+type ContainerdRegistryConfigTLS struct {
+}
+
+type ContainerdRegistryConfigAuth struct {
+	Username      string `json:"username,omitempty"`
+	Password      string `json:"password,omitempty"`
```

Comment:
I don't like putting plain-text credentials in the configuration file. However, at this point, I don't see a better solution. I think we should at least ensure it's possible to source those two fields from the credentials file.